lib/ansible/modules/unarchive.py

```diff
@@ -978,7 +978,8 @@ def __init__(self, src, b_dest, file_args, module):
 class ZipZArchive(ZipArchive):
     def __init__(self, src, b_dest, file_args, module):
         super(ZipZArchive, self).__init__(src, b_dest, file_args, module)
-        self.zipinfoflag = '-Z'
+        # NOTE: adds 'l', which is default on most linux but not all implementations
+        self.zipinfoflag = '-Zl'
```

Comment:
Once this change is confirmed as working, it should be made in a separate PR (with a changelog entry), so it can be backported as needed.

Reply:
that was the plan!

Reply:
#82873